ddtrace/llmobs/_trace_processor.py

```diff
@@ -57,11 +58,17 @@ def process_trace(self, trace: List[Span]) -> Optional[List[Span]]:
 
     def submit_llmobs_span(self, span: Span) -> None:
         """Generate and submit an LLMObs span event to be sent to LLMObs."""
+        span_event = None
         try:
             span_event = self._llmobs_span_event(span)
             self._span_writer.enqueue(span_event)
         except (KeyError, TypeError):
             log.error("Error generating LLMObs span event for span %s, likely due to malformed span", span)
+        finally:
+            if not span_event:
+                return
+            if self._evaluator_runner:
+                self._evaluator_runner.enqueue(span_event)
```

Comment:
Great point. I can see that there may be instances where the write fails for some reason but we still submit an eval metric. 

I don't think a write failure for a span should block prevent an evaluation metric from being submitted though, since users want to have as much data as possible for "average faithfulness scores over a week" which doesn't depend on the eval metric being tied to a span

Reply:
makes sense to me, thanks for the clarification!